packages/lesswrong/components/posts/PostsPage/RejectionNotice.tsx

```diff
@@ -0,0 +1,99 @@
+import React, { useState, useMemo } from 'react';
+import { defineStyles, useStyles } from '@/components/hooks/useStyles';
+import { ContentItemBody } from "../../contents/ContentItemBody";
+import ContentStyles from "../../common/ContentStyles";
+
+const styles = defineStyles('RejectionNotice', (theme: ThemeType) => ({
+  root: {
+    ...theme.typography.contentNotice,
+    ...theme.typography.commentStyle,
+    padding: 12,
+    paddingBottom: 4,
+    paddingRight: 24,
+    maxWidth: 600,
+    marginBottom: 30,
+    backgroundColor: theme.palette.grey[100],
+    cursor: 'pointer',
+  },
+  headerText: {
+    fontSize: 14,
+    color: theme.palette.grey[900],
+  },
+  summaryContent: {
+    ...theme.typography.commentStyle,
+    ...theme.typography.body2,
+    '& li': {
+      ...theme.typography.commentStyle,
+      ...theme.typography.body2,
+      marginBottom: 0,
+    },
+  },
+  expandedReason: {
+    '& ul, & ol': { paddingLeft: 20 },
+    '& li': { marginBottom: 16 },
+  },
+  collapseButton: {
+    marginTop: -8,
+    fontSize: 12,
+    color: theme.palette.primary.main,
+    cursor: 'pointer',
+    textAlign: 'center',
+  },
+}));
+
+// Extracts a summary from HTML rejection reasons:
+// - For lists (<ul>/<ol>): extracts text from each <li> item
+// - For paragraphs: extracts the first <p> content
+// - Text extraction includes multiple sentences if they are all bold, or just the first sentence if they are not. (Since we often use bold to convey multi-sentence-summaries)
+
+function extractLeadingText(html: string): string {
+  const trimmed = html.trim();
+  const boldMatch = trimmed.match(/^<(strong|b)>([\s\S]*?)<\/\1>/i);
+  if (boldMatch) return boldMatch[2].replace(/<[^>]*>/g, '').trim();
+  const plainText = trimmed.replace(/<[^>]*>/g, ' ').replace(/\s+/g, ' ').trim();
+  const sentenceMatch = plainText.match(/^[^.!?]*[.!?]/);
+  return sentenceMatch ? sentenceMatch[0].trim() : plainText;
+}
+
+function extractSummary(html: string): string[] {
+  const trimmedHtml = html.trim().replace(/^<p>\s*<\/p>/gi, '');
+  if (/^<[ou]l/i.test(trimmedHtml)) {
+    return (trimmedHtml.match(/<li[^>]*>([\s\S]*?)<\/li>/gi) || [])
+      .map(li => extractLeadingText(li.replace(/<\/?li[^>]*>/gi, '')))
+      .filter(Boolean);
+  }
+  const firstParagraph = trimmedHtml.match(/<p[^>]*>([\s\S]*?)<\/p>/i);
+  const leading = extractLeadingText(firstParagraph ? firstParagraph[1] : trimmedHtml);
+  return leading ? [leading] : [];
+}
```

Comment:
This is IMO a pretty deranged way to be doing this, when we have `parseDocumentFromString` and then the ability to query the parsed html properly.